### PR TITLE
Task Management ADR

### PR DESCRIPTION
**Pull request by ssatagopan**
# Evaluate and select task management system(https://github.com/OHS-Hosting-Infrastructure/infrastructure/issues/31)

Changes proposed in this pull request:

- Initial version of ADR with proposed selection for task management SW

### CODE REVIEW

**Review by kilbergr — COMMENTED** [review]
So I think this is a reasonable outcome, but I do wonder why the Truss default, Pivotal Tracker, wasn't on this list. Is there anything that makes it an obvious no go? I am fine with JIRA--just wanted to check in.
I would definitely find out what the price estimate is.

**Comment by ssatagopan** [comment]
> So I think this is a reasonable outcome, but I do wonder why the Truss default, Pivotal Tracker, wasn't on this list. Is there anything that makes it an obvious no go? I am fine with JIRA--just wanted to check in.
> I would definitely find out what the price estimate is.

That's a very good point. I didn't include it only b/c I didn't think of it.

**Comment by ssatagopan** [comment]
Added PivotalTracker to ADR. Still proposing JIRA, mainly due to more flexibility to accommodate different workflows as well as ticket attributes and timeline/planning functions

**Review by rahearn — APPROVED** [review]
LGTM - there is one extra bullet on the list of options that could be cleaned up, but it's also not hurting anything

**Review by eeeady — APPROVED** [review]
Generally I'm happy with this. I do think before we execute we need to figure out the costs for using JIRA and the overhead for getting it ATO'ed.  I _highly_ suggest noting that JIRA we are hoping for will be JIRA Cloud. If we have to self host, there will be be year over year maintenance overhead that makes JIRA less appealing.

File: docs/adr/004-task-management-sw.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+### JIRA
+
+Solution made for agile teams. [JIRA](https://www.atlassian.com/software/jira)
+
+* Good, because people know how to use it already
+* Good, because it's already being used by other teams so should be easier to integrate with
+* Good, because it meets all our requirements
+* Bad, because it's not FedRAMP authorized so it will take longer for an ATO
+* … <!-- numbers of pros and cons can vary -->
```
Comment (kilbergr) [inline docs/adr/004-task-management-sw.md:124]: I would also add that it's a complex product that requires lots of configuration in my experience.

File: docs/adr/004-task-management-sw.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+### PivotalTracker
+
+Out of the box solution for agile teams. [PivotalTracker](https://www.pivotaltracker.com/)
+
+* Good, because it's TRUSS's task management tool of record
+* Good, because the built-in workflows accommodate the basic workflows team would need
+* Good, because it has integrations to GitHub, JIRA, Bitbucket, Slack
+* Bad, because it does not have flexibility to accommodate changes in workflow or additional properties in tickets
+* Bad, because it does not seem to have built-in time line or planning functions
```
Comment (kilbergr) [inline docs/adr/004-task-management-sw.md:134]: It has burn down charts I believe. Is this not what you're describing?

**Review by kilbergr — CHANGES_REQUESTED** [review]
I agree with Eady--if we have to self host JIRA, I would push for Pivotal instead because nobody (on this team) has time for that. I would also find out what the costs are. If cost is prohibitive, that will cancel it out as an option.

File: docs/adr/004-task-management-sw.md (hunk 2 above)
Comment (ssatagopan) [reply docs/adr/004-task-management-sw.md:134]: > It has burn down charts I believe. Is this not what you're describing?

No, I'm talking about being able to track expected completion dates/actual completion dates and roadmapping tools

File: docs/adr/004-task-management-sw.md (hunk 1 above)
Comment (ssatagopan) [reply docs/adr/004-task-management-sw.md]: > I would also add that it's a complex product that requires lots of configuration in my experience.

Added

**Comment by ssatagopan** [comment]
> So I think this is a reasonable outcome, but I do wonder why the Truss default, Pivotal Tracker, wasn't on this list. Is there anything that makes it an obvious no go? I am fine with JIRA--just wanted to check in.
> I would definitely find out what the price estimate is.

Added note that we are evaluating JIRA Cloud

File: docs/adr/004-task-management-sw.md (hunk 2 above)
Comment (kilbergr) [reply docs/adr/004-task-management-sw.md:133]: Gotcha!

**Comment by kilbergr** [comment]
So--do you think we need the cost to make a choice? Or no?
LMK what you think and I'll approve if you think it's unnecessay.

**Comment by ssatagopan** [comment]
> So--do you think we need the cost to make a choice? Or no?
> LMK what you think and I'll approve if you think it's unnecessary.

I think we do need cost to make a choice, so I would leave it unmerged until we decide. Still waiting on ACF JIRA owner to communicate that. I should have cost for ServiceNow in a couple days so we could compare with that at least. I've reached out to Pivotal as well for a quote.

**Comment by ssatagopan** [comment]
@kilbergr Based on our discussion during sprint planning, if you're ok with my updates, let me know and I'll go ahead and merge PR but keep issue open

**Review by kilbergr — APPROVED** [review]
Works for me !

**Pull request merged**